library/EventListener/MetadataOperations.php

```diff
@@ -107,8 +109,20 @@ public function search(EventInterface $event) {
             'to' => $params->get('to'),
         ];
 
+        if ($queryParams['page'] < 1) {
+            throw new RuntimeException('Invalid param. "page" must be a positive number.', 400);
+        }
+
+        if ($queryParams['limit'] < 1) {
+            throw new RuntimeException('Invalid param. "limit" must be a positive number.', 400);
+        }
+
         // Parse the query JSON and transform it to an AST
-        $ast = DslParser::parse($metadataQuery);
+        try {
+            $ast = DslParser::parse($metadataQuery);
+        } catch (\Exception $e) {
+            throw new RuntimeException('Invalid metadata query', 400);
```

Comment:
I agree. That is a better approach. The reason for throwing a runtime exception is that it leads to a proper error response from Imbo, but as long as we find a way of doing that from an InvalidArgument exception that is fine.

Might be worth looking into using the InvalidArgument exception from Imbo core - I think that leads to a proper response instead of a 500 error.